src/openforms/submissions/templates/report/submission_report.html

```diff
@@ -101,11 +101,19 @@ <h2 class="subtitle">{{ submission_step_node.render }}</h2>
             {% endif %}
 
             {% if config.ask_privacy_consent and report.submission.privacy_policy_accepted %}
```

Comment:
this now only looks at the global configuration (via `config`) but doesn't take the form configuration into account. There is a `Form.get_statement_checkbox_required` method that you should use for this (it's also what the validators in the serializer use, see https://github.com/open-formulieren/open-forms/blob/master/src/openforms/submissions/api/validators.py#L77)